From stern@rowland.harvard.edu Fri Apr 14 13:42:10 2006
Date: Fri, 14 Apr 2006 16:42:03 -0400 (EDT)
From: Alan Stern <stern@rowland.harvard.edu>
To: Greg KH <greg@kroah.com>, David Brownell <david-b@pacbell.net>
Subject: USB: net2280: send 0-length packets for ep0
Message-ID: <Pine.LNX.4.44L0.0604141640010.5320-100000@iolanthe.rowland.org>

This patch (as669) fixes a bug in the net2280 driver.  Now it will
properly send zero-length packets on ep0 until the control status stage
occurs.

Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/net2280.c |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- gregkh-2.6.orig/drivers/usb/gadget/net2280.c
+++ gregkh-2.6/drivers/usb/gadget/net2280.c
@@ -2280,9 +2280,7 @@ static void handle_ep_small (struct net2
 		/* if we wrote it all, we're usually done */
 		if (req->req.actual == req->req.length) {
 			if (ep->num == 0) {
-				/* wait for control status */
-				if (mode != 2)
-					req = NULL;
+				/* send zlps until the status stage */
 			} else if (!req->req.zero || len != ep->ep.maxpacket)
 				mode = 2;
 		}
